route-rs-runtime/src/link/mod.rs

```diff
@@ -12,8 +12,8 @@ pub use self::sync_link::*;
 /// useful for creating queues in the router, buffering, and creating `Task` boundries that can be processed on
 /// different threads, or even different cores. Before packets are placed into the queue to be output, they are run
 /// through the element defined process function, often performing some sort of transformation.
-mod async_link;
-pub use self::async_link::*;
+mod queue_link;
```

Comment:
nit: I know that the queue structure is exposed via builder method, but if I'm a user I'm not quite sure what the "queue" gets me. This link can output to multiple egressors and wait for long-running processes to complete (which `SyncLink` can't),  but that's not obvious from the name `QueueLink`. I'm also not a big fan of `AsyncLink` either though. I'm cool with this for now, but we should definitely think hard about the names of our primitive links before any kind of release. We might notice some more API UX problems like, "if there's a SyncLink why is there no AsyncLink?", that might be more noticeable after finishing the NAT and having other people play around with the code. _user interviews??_

To put some more names into the ring: `BufferedLink` and `UnbufferedLink`. They're complements, and you can kindof intuit their properties based on the names?

Reply:
Hmm,  I do agree not having a Sync/Async complement is an issue. Not sure I like buffered link,   FYR, the queuelink is still 1-1.  If you want multi-output, then you need a CloneLink.   I thought about supporting the queuelink as multi-output, but figured it would be better to just put a SyncLink->CloneLink to get the desired effect at minimal cost.  

Crazytown thought. What if the only Link that can take elements is the SyncLink?  QueueLink or BufferLink is for...buffers?  We could rename Clone to Fork, so there are obvious Fork/Join complements.  Classify would remain its own special link

Reply:
https://gph.is/g/E3geJNn

Reply:
Going to leave this thought for meetup. We should make the choice as a group, since it basically will cement the design philosophy for our links overall.